### PR TITLE
extract rpc and mpt into crates

### PR DESCRIPTION
**Pull request by CHr15F0x**
`pathfinder_lib` incremental build times on my PC:
- `0.4.0` ~2.7s
- now ~0.7s

I really didn't want to put the mpt stuff into `common` so unfortunately it landed in a separate crate.

The cairo stuff had to go into the rpc crate as they're just too interlinked.

### CODE REVIEW

**Review by sergey-melnychuk — APPROVED** [review]
LGTM

**Review by kkovaacs — APPROVED** [review]
LGTM

File: crates/rpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+serde = { version = "1.0.130", features = ["derive"] }
+serde_json = { version = "1.0.68", features = ["arbitrary_precision", "raw_value"] }
+serde_with = "1.9.4"
+stark_hash = { path = "../stark_hash" }
+starknet-gateway-client = { path = "../gateway-client" }
+starknet-gateway-types = { path = "../gateway-types" }
+thiserror = "1.0.30"
+tokio = "1.11.0"
+tracing = "0.1.31"
+web3 = "0.18.0"
```
Comment (Mirko-von-Leipzig) [inline crates/rpc/Cargo.toml:35]: There should be no `web3` left
```suggestion
```

**Review by Mirko-von-Leipzig — CHANGES_REQUESTED** [review]
LGTM apart from the lockfile having a bunch of duplicate packages in it now. I suspect its probably just the `web3` extra that snuck it.

File: crates/rpc/Cargo.toml (hunk 1 above)
Comment (CHr15F0x) [reply crates/rpc/Cargo.toml:35]: It could be a side effect of the recent rebase :eyes:

**Review by Mirko-von-Leipzig — APPROVED** [review]
Awesome lockfile looks much better!

**Pull request merged**